packages/backend-data/src/factory.ts

```diff
@@ -149,7 +156,38 @@ class DataGenerator implements ConstructContainerEntryGenerator {
         ? this.props.schema.schemas
         : [this.props.schema];
 
-      schemas.forEach((schema) => {
+      // get the branch name and use the imported table map for that key
+      // use the sandbox key when the branch name is not available (e.g. in the sandbox deployment)
+      const amplifyBranchName =
+        scope.node.tryGetContext('amplifyEnvironmentName') ?? 'sandbox';
+      // ensure all branch names are unique
+      if (this.props.migratedAmplifyGen1DynamoDbTableMap) {
+        // Remove this warning for GA
+        Annotations.of(scope).addWarning(
+          'migratedAmplifyGen1DynamoDbTableMap is experimental and is not recommended for production use. This functionality may be changed or removed without warning.',
+        );
```

Comment:
we should annotate the property with `@experimental` then.

Reply:
This isn't a valid JSDoc tag according to eslint.

```
156:1  error  Invalid JSDoc tag name "experimental"  jsdoc/check-tag-names
```

Looks like it is a feature request, but has not been implemented. https://github.com/microsoft/TypeScript/issues/56808